src/main/java/org/jabref/logic/pdf/search/indexing/IndexingTaskManager.java

```diff
@@ -0,0 +1,188 @@
+package org.jabref.logic.pdf.search.indexing;
+
+import java.util.List;
+
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+
+import org.apache.lucene.store.Directory;
+
+/**
+ * Wrapper around {@link PdfIndexer} to execute all operations in the background.
+ */
+public class IndexingTaskManager extends BackgroundTask<Void> {
```

Comment:
void cannot be used as a type parameter since it's not a class, therefore you need to use Void. This is so it is clear that the return value is not used.

Reply:
Ah ok.  Special behavior of void. I'm probably thinking to formalistic.